src/echo.js

```diff
@@ -23,12 +27,23 @@ function _echo(opts, messages) {
   // allow strings starting with '-', see issue #20
   messages = [].slice.call(arguments, opts ? 0 : 1);
 
-  if (messages[0] === '-e') {
-    // ignore -e
+  var option = ['-e', '-n', '-ne', '-en'].indexOf(messages[0]);
+  var output;
+
+  if (option >= 0) {
+    // ignore options
     messages.shift();
+    output = format.apply(null, messages);
+    if (option === 0) {
+      // add newline if -n is not passed
+      output += '\n';
+    }
+  } else {
+    output = format.apply(null, messages) + '\n';
   }
 
-  console.log.apply(console, messages);
```

Comment:
Let's only use `process.stdout.write` for when `-n` is passed. Calling `process.exit` immediately after `process.stdout.write` has some issues.